linter-rules/src/test/groovy/software/amazon/nextflow/rules/healthomics/InputS3FileRuleTest.groovy

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.nextflow.rules.healthomics
+
+
+import org.codenarc.rule.AbstractRuleTestCase
+import org.junit.jupiter.api.Test
+
+/*
+ * Tests for InputFileIsS3Rule
+ *
+ * @author Jesse Marks
+ */
+
+class InputS3FileRuleTest extends AbstractRuleTestCase<InputS3FileRule> {
+
+    @Override
+    protected InputS3FileRule createRule() {
+        new InputS3FileRule()
+    }
+
+    @Test
+    void RuleProperties() {
+        assert rule.priority == 1
+        assert rule.name == 'InputFileIsS3'
+    }
+
+    @Test
+    void filePath_NoViolations() {
+       final SOURCE = '''
+            process bar {
+                input:
+                  path "s3://my-input-bucket/data/input1.txt"
+                  path "s3://my-input-bucket/data/input2.txt"
+
+                """
+                echo "All input files are valid S3 paths."
+                """
+            }
+        '''
+        assertNoViolations(SOURCE)
+    }
+
+    @Test
+    void filePath_TooManyArgsViolations() {
+       final SOURCE = '''
+            process bar {
+                input:
+                  path 's3://my-input-bucket/data/input1.txt', 's3://my-input-bucket/data/input2.txt'
```

Comment:
I think this is correct but there are ways in which Path can be a list of files. https://www.nextflow.io/docs/latest/process.html#multiple-input-files